### PR TITLE
fix(mme): AMF module warning cleanup#8985

### PR DESCRIPTION
**Pull request by sreedharkumartn**
Signed-off-by: sreedharkumartn <sreedhar.kumar@wavelabs.ai>

## Summary
Added fixes for AMF warnings.

## Test Plan
 - Verified basic sanity with UERANSIM.
 - Please find attached logs and pcap snap for basic registration and PDU Session setup
 
Logs:
[mme_warning_cleanup.log](https://github.com/magma/magma/files/7204550/mme_warning_cleanup.log)

Pcap snap:
image

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/c/core/oai/tasks/nas5g/src/ies/M5GMMCause.cpp [hunk 1]
```diff
@@ -47,14 +46,14 @@ int M5GMMCauseMsg::EncodeM5GMMCauseMsg(
   if (iei > 0) {
     *(buffer + encoded) = m5gmm_cause->iei;
     CHECK_IEI_ENCODER((unsigned char) iei, m5gmm_cause->iei);
-    MLOG(MDEBUG) << "In EncodeM5GMMCauseMsg: iei = " << hex
-                 << int(*(buffer + encoded)) << endl;
+    MLOG(MDEBUG) << "In EncodeM5GMMCauseMsg: iei = " << std::hex
+                 << int(*(buffer + encoded)) << std::endl;
```
Comment (lionelgo) [inline lte/gateway/c/core/oai/tasks/nas5g/src/ies/M5GMMCause.cpp:50]: We should harmonize hex or dec to trace iei

File: lte/gateway/c/core/oai/tasks/nas5g/src/ies/M5GAuthenticationParameterAUTN.cpp [hunk 2]
```diff
@@ -42,7 +41,7 @@ int AuthenticationParameterAUTNMsg::EncodeAuthenticationParameterAUTNMsg(
   if (iei > 0) {
     CHECK_IEI_ENCODER((unsigned char) iei, autn->iei);
     *buffer = iei;
-    MLOG(MDEBUG) << "In EncodeAuthenticationParameterAUTNMsg: iei" << hex
+    MLOG(MDEBUG) << "In EncodeAuthenticationParameterAUTNMsg: iei" << std::hex
```
Comment (lionelgo) [inline lte/gateway/c/core/oai/tasks/nas5g/src/ies/M5GAuthenticationParameterAUTN.cpp:44]: IEI here in hex

File: lte/gateway/c/core/oai/tasks/nas5g/src/ies/M5GAuthenticationFailureIE.cpp [hunk 3]
```diff
@@ -29,8 +28,8 @@ int M5GAuthenticationFailureIE::DecodeM5GAuthenticationFailureIE(
   if (iei > 0) {
     m5g_auth_failure_ie->iei = *(buffer + decoded);
     CHECK_IEI_DECODER((unsigned char) iei, m5g_auth_failure_ie->iei);
-    MLOG(MDEBUG) << "In DecodeM5GAuthenticationFailureIE: iei = " << dec
-                 << int(m5g_auth_failure_ie->iei) << endl;
+    MLOG(MDEBUG) << "In DecodeM5GAuthenticationFailureIE: iei = " << std::dec
```
Comment (lionelgo) [inline lte/gateway/c/core/oai/tasks/nas5g/src/ies/M5GAuthenticationFailureIE.cpp]: IEI here in dec

File: lte/gateway/c/core/oai/tasks/nas5g/src/ies/M5GSMobileIdentity.cpp [hunk 4]
```diff
@@ -83,22 +82,22 @@ int M5GSMobileIdentityMsg::DecodeGutiMobileIdentityMsg(
   decoded++;
   guti->tmsi4 = *(buffer + decoded);
   decoded++;
-  MLOG(MDEBUG) << "   Odd/Even Indecation = " << dec << int(guti->odd_even)
+  MLOG(MDEBUG) << "   Odd/Even Indecation = " << std::dec << int(guti->odd_even)
```
Comment (lionelgo) [inline lte/gateway/c/core/oai/tasks/nas5g/src/ies/M5GSMobileIdentity.cpp]: typo indication

File: lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp [hunk 5]
```diff
@@ -723,14 +723,16 @@ int amf_app_handle_pdu_session_accept(
...
+        LOG_AMF_APP, "ue context not found for the ue_id : " AMF_UE_NGAP_ID_FMT,
+        ue_id);
     return M5G_AS_FAILURE;
   }
 
   smf_ctx = amf_smf_context_exists_pdu_session_id(
       ue_context, pdu_session_resp->pdu_session_id);
   if (!smf_ctx) {
-    OAILOG_ERROR(LOG_AMF_APP, "Smf context is not exist UE ID: %u", ue_id);
+    OAILOG_ERROR(LOG_AMF_APP, "Smf context is not exist UE ID: %lu", ue_id);
```
Comment (lionelgo) [inline lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp]: we could use AMF_UE_NGAP_ID_FMT.

**Review by lionelgo — CHANGES_REQUESTED** [review]

File: lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp (hunk 5 above)
Comment (sreedharkumartn) [reply lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp]: Addressed the review comment

File: lte/gateway/c/core/oai/tasks/nas5g/src/ies/M5GSMobileIdentity.cpp (hunk 4 above)
Comment (sreedharkumartn) [reply lte/gateway/c/core/oai/tasks/nas5g/src/ies/M5GSMobileIdentity.cpp]: Addressed the review comment

File: lte/gateway/c/core/oai/tasks/nas5g/src/ies/M5GMMCause.cpp (hunk 1 above)
Comment (sreedharkumartn) [reply lte/gateway/c/core/oai/tasks/nas5g/src/ies/M5GMMCause.cpp:50]: hex format is a better format to trace IEI, so we can compare pcap buffer with trace directly.

File: lte/gateway/c/core/oai/tasks/nas5g/src/ies/M5GAuthenticationParameterAUTN.cpp (hunk 2 above)
Comment (sreedharkumartn) [reply lte/gateway/c/core/oai/tasks/nas5g/src/ies/M5GAuthenticationParameterAUTN.cpp:44]: hex format is a better format to trace IEI, so we can compare pcap buffer with trace directly.

File: lte/gateway/c/core/oai/tasks/nas5g/src/ies/M5GAuthenticationFailureIE.cpp (hunk 3 above)
Comment (sreedharkumartn) [reply lte/gateway/c/core/oai/tasks/nas5g/src/ies/M5GAuthenticationFailureIE.cpp]: We have changed IEI format to hex.

**Comment by sreedharkumartn** [comment]
verified UERANSIM.
please find logs and pcap attached.
[mme_warning_cleanup.log](https://github.com/magma/magma/files/7204364/mme_warning_cleanup.log)
image

**Comment by rdefosse** [comment]
@sreedharkumartn 

OAI CI build failed --> `https://jenkins-oai.eurecom.fr/job/MAGMA-MME-production/6696/artifact/build_results_magma_oai_mme.html`

```
`lte/gateway/c/core/oai/tasks/nas5g/src/ies/M5GPDUSessionIdentity.cpp`  line 32 | Error | fatal | 'endl' was not declared in this scope
```

**Comment by sreedharkumartn** [comment]
I did some mistake and messed this review. I raised separate PR https://github.com/magma/magma/pull/9238.
I addressed all the review comments from @lionelgo and @rdefosse. I updated the PR #9238 . 
So, I am discarding this PR.

**Pull request closed without merging**